### PR TITLE
native query editor metabot suggestions

### PR DESCRIPTION
**Pull request by alxnddr**
### Description

Integrate Metabot suggestions right into the native query editor.

Screen Shot 2023-04-06 at 6 38 46 PM

### How to verify

- Enable Metabot in the admin settings http://localhost:3000/admin/settings/metabot
- Go to the native query editor and in the editor sidebar open the prompt input
- Try generating queries

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/29875)

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/NativeQueryEditorPrompt/NativeQueryEditorPrompt.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    [generateQuery],
+  );
+
+  return (
+    <NativeQueryEditorPromptRoot>
+      {loading ? <LoadingSpinner size={20} /> : <Icon name="insight" />}
+
+      {error != null ? (
+        <ErrorContent
+          error={(error as any)?.data?.message}
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/NativeQueryEditorPrompt/NativeQueryEditorPrompt.tsx:70]: `getResponseErrorMessage(error)`

File: frontend/src/metabase/query_builder/components/NativeQueryEditor.jsx [hunk 2]
```diff
@@ -512,10 +514,36 @@ class NativeQueryEditor extends Component {
...
+
+  handleQueryGenerated = queryText => {
+    this.handleQueryUpdate(queryText);
+    this._editor.focus();
+  };
+
+  isPromptInputVisible = () => {
+    const { canUsePromptInput } = this.props;
+    const database = this.props.query.database();
+    const isSupported =
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/NativeQueryEditor.jsx:536]: I believe we should do something similar in metabot components. Could we create a common function for this?

**Review by ranquild — CHANGES_REQUESTED** [review]
@alxnddr I wasn't able to see the prompt from the screenshot. `isPromptInputVisible` seems to be always false. Could you take a look?

File: src/metabase/metabot/util.clj [hunk 3]
```diff
@@ -151,6 +153,72 @@
...
+  "For a field, determine any potential enumerated values."
+  [{table-name :name :as _table} {field-name :name field-id :id :keys [base_type]}]
+  (when-let [values (and
+                     (not= :type/Boolean base_type)
+                     (:values (t2/select-one FieldValues :field_id field-id)))]
+    (when (< (count values) (metabot-settings/enum-cardinality-threshold))
+      (format "create type %s_%s_t as enum %s;"
+             table-name
+             field-name
+             (str/join ", " (map (partial format "'%s'") values))))))
```
Comment (metamben) [inline src/metabase/metabot/util.clj:166]: Indentation.

File: src/metabase/metabot/util.clj [hunk 4]
```diff
@@ -151,6 +153,72 @@
...
+                        (cond-> [column-name
+                                 (if (enums column-name)
+                                   (format "%s_%s_t" table-name column-name)
+                                   database_type)]
+                          database_required
+                          (conj [:not nil]))))
+        primary-keys [[(into [:primary-key]
+                             (->> fields
+                                  (filter (comp #{:type/PK} :semantic_type))
+                                  (map :name)))]]
```
Comment (metamben) [inline src/metabase/metabot/util.clj:193]:
```suggestion
        primary-keys [[(into [:primary-key]
                             (comp (filter (comp #{:type/PK} :semantic_type))
                                   (map :name))
                             fields)]]
```

**Review by metamben — APPROVED** [review]
Back end LGTM

**Review by ranquild — APPROVED** [review]

File: src/metabase/metabot/util.clj [hunk 5]
```diff
@@ -151,6 +153,72 @@
 (defn- add-model-json-summary [database]
   (assoc database :model_json_summary (models->json-summary database)))
 
+(defn- field->pseudo-enums
```
Comment (camsaul) [inline src/metabase/metabot/util.clj:156]: https://github.com/metabase/metabase/wiki/Metabase-Clojure-Style-Guide#make-things-private

**Pull request merged**